framework/tools/generators.py

```diff
@@ -14,3 +19,45 @@ def generate_string(length: int, additional_characters: list = None) -> str:
         result += additional_characters
 
     return "".join(result)
+
+
+def generate_password(length: int) -> str:
+    """
+     Function generates and returns random generated password.
+     Args:
+         length (int): length of password
+    """
+    letters = string.ascii_letters
+    digits = string.digits
+    special_chars = "!@%*$&"
+    all_chars = digits + special_chars + letters
+    password = ''.join(random.choice(all_chars) for _ in range(length))
+
+    """Digits and letters is required char in password. This statement add 1 digits and letter to the password" \
+       if it absent during generation random password """
+    if all(char not in digits for char in password):
+        random_index = random.randint(0, length - 1)
+        password = password[:random_index] + random.choice(digits) + password[random_index:]
+    if all(char not in letters for char in password):
+        random_index = random.randint(0, length - 1)
+        password = password[:random_index] + random.choice(letters) + password[random_index:]
+
+    return password
+
+
+def generate_user_data(password_len: int = None, length_first_name: int = None, length_last_name: int = None) -> dict:
```

Comment:
Please:
1) naming to a single view: password_length, first_name_length, etc. Or other
2) default not None. Else error TypeError

Reply:
Fixed